packages/components/src/button/types.ts

```diff
@@ -0,0 +1,125 @@
+/**
+ * External dependencies
+ */
+import type { ReactNode } from 'react';
+
+/**
+ * Internal dependencies
+ */
+import type { Props as IconProps } from '../icon';
+
+export type ButtonProps = {
+	/**
+	 * The button's children.
+	 */
+	children?: ReactNode;
+	/**
+	 * An accessible description for the button.
+	 */
+	describedBy?: string;
+	/**
+	 * Whether the button is disabled.
+	 * If `true`, this will force a `button` element to be rendered.
+	 */
+	disabled?: boolean;
+	/**
+	 * Whether the button is focused.
+	 */
+	focus?: boolean;
+	/**
+	 * If provided, renders `a` instead of `button`.
+	 */
+	href?: string;
+	/**
+	 * If provided, renders an Icon component inside the button.
+	 */
+	icon?: IconProps< unknown >[ 'icon' ];
+	/**
+	 * If provided with `icon`, sets the position of icon relative to the `text`.
+	 *
+	 * @default 'left'
+	 */
+	iconPosition?: 'left' | 'right';
+	/**
+	 * If provided with `icon`, sets the icon size.
+	 * Please refer to the Icon component for more details regarding
+	 * the default value of its `size` prop.
+	 */
+	iconSize?: IconProps< unknown >[ 'size' ];
+	/**
+	 * Indicates activity while a action is being performed.
+	 */
+	isBusy?: boolean;
+	/**
+	 * Renders a red text-based button style to indicate destructive behavior.
+	 */
+	isDestructive?: boolean;
+	/**
+	 * Renders a pressed button style.
+	 */
+	isPressed?: boolean;
+	/**
+	 * Decreases the size of the button.
+	 */
+	isSmall?: boolean;
+	/**
+	 * Sets the `aria-label` of the component, if none is provided.
+	 * Sets the Tooltip content if `showTooltip` is provided.
+	 */
+	label?: string;
+	/**
+	 * If provided with `showTooltip`, appends the Shortcut label to the tooltip content.
+	 * If an object is provided, it should contain `display` and `ariaLabel` keys.
+	 */
+	shortcut?: string | { display: string; ariaLabel: string };
+	/**
+	 * If provided, renders a Tooltip component for the button.
+	 */
+	showTooltip?: boolean;
+	/**
+	 * If provided with `href`, sets the `target` attribute to the `a`.
+	 */
+	target?: string;
+	/**
+	 * If provided, displays the given text inside the button. If the button contains children elements, the text is displayed before them.
+	 */
+	text?: string;
+	/**
+	 * If provided with `showTooltip`, sets the position of the tooltip.
+	 * Please refer to the Tooltip component for more details regarding the defaults.
+	 */
+	tooltipPosition?: string;
```

Comment:
Let's reference the types for the underlying component:

```suggestion
	tooltipPosition?: PopoverProps[ 'position' ];
```

Reply:
Thanks, committed in https://github.com/WordPress/gutenberg/pull/46997/commits/916f9ac45ae0d40b844bc58884828e9077601d40